global_segment_map_node/src/iodb_controller.cpp

```diff
@@ -0,0 +1,591 @@
+
+  // Determine feature integrator parameters.
+  // TODO(ntonci): Make rosparam if there are any.
+  FeatureIntegrator::Config feature_integrator_config;
+  feature_integrator_.reset(
+      new FeatureIntegrator(feature_integrator_config, feature_layer_.get()));
+
+  node_handle_private_->param<bool>("publish_feature_blocks_marker",
+                                    publish_feature_blocks_marker_,
+                                    publish_feature_blocks_marker_);
+}
+
+// TODO(ntonci): Move all sub, pub, etc., and helpers to inline header.
+void IodbController::subscribeFeatureTopic(ros::Subscriber* feature_sub) {
+  CHECK_NOTNULL(feature_sub);
+  std::string feature_topic = "/rgbd_feature_node/features";
+  node_handle_private_->param<std::string>("feature_topic", feature_topic,
+                                           feature_topic);
+
+  // Large queue size to give slack to the
+  // pipeline and not lose any messages.
+  constexpr int kFeatureQueueSize = 2000;
+  *feature_sub = node_handle_private_->subscribe(
+      feature_topic, kFeatureQueueSize, &IodbController::featureCallback, this);
+}
+
+void IodbController::advertiseFeatureBlockTopic(
+    ros::Publisher* feature_block_pub) {
+  CHECK_NOTNULL(feature_block_pub);
+  std::string feature_block_topic = "feature_block_array";
+  node_handle_private_->param<std::string>(
+      "feature_block_topic", feature_block_topic, feature_block_topic);
+
+  constexpr int kFeatureBlockQueueSize = 2000;
+  *feature_block_pub =
+      node_handle_private_->advertise<visualization_msgs::MarkerArray>(
+          feature_block_topic, kFeatureBlockQueueSize, true);
+  feature_block_pub_ = feature_block_pub;
+}
+
+void IodbController::advertiseSegmentGsmUpdateTopic(
+    ros::Publisher* segment_gsm_update_pub) {
+  CHECK_NOTNULL(segment_gsm_update_pub);
+  std::string segment_gsm_update_topic = "gsm_update";
+  node_handle_private_->param<std::string>("segment_gsm_update_topic",
+                                           segment_gsm_update_topic,
+                                           segment_gsm_update_topic);
+  // TODO(ff): Reduce this value, once we know some reasonable limit.
+  constexpr int kGsmUpdateQueueSize = 2000;
+  *segment_gsm_update_pub =
+      node_handle_private_->advertise<modelify_msgs::GsmUpdate>(
+          segment_gsm_update_topic, kGsmUpdateQueueSize, true);
+  segment_gsm_update_pub_ = segment_gsm_update_pub;
+}
+
+void IodbController::advertiseSceneGsmUpdateTopic(
+    ros::Publisher* scene_gsm_update_pub) {
+  CHECK_NOTNULL(scene_gsm_update_pub);
+  std::string scene_gsm_update_topic = "scene";
+  node_handle_private_->param<std::string>(
+      "scene_gsm_update_topic", scene_gsm_update_topic, scene_gsm_update_topic);
+  constexpr int kGsmSceneQueueSize = 1;
+
+  *scene_gsm_update_pub =
+      node_handle_private_->advertise<modelify_msgs::GsmUpdate>(
+          scene_gsm_update_topic, kGsmSceneQueueSize, true);
+  scene_gsm_update_pub_ = scene_gsm_update_pub;
```

Comment:
Can you not just pass in controller->scene_gsm_update_pub_ here and in the other advertise methods?
I did see, that this was done like this before, but I think we can save constructing the Publisher outside, which would just be slightly nicer.